### ISSUE TITLE
Support Spring Cloud 2024.0 (Spring Boot 3.4, Spring Framework 6.2)

### ISSUE THREAD

**Issue by breun** [GoogleCloudPlatform/spring-cloud-gcp#3320]
Maybe this is already on your radar, but the following Spring ecosystem updates are scheduled for Q4 2024:

Release candidates
* ✅ October 17: [Spring Framework 6.2.0-RC2](https://github.com/spring-projects/spring-framework/releases/tag/v6.2.0-RC2)
* ✅ October 24: [Spring Framework 6.2.0-RC3](https://github.com/spring-projects/spring-framework/releases/tag/v6.2.0-RC3)
* ✅ October 26: [Spring Boot 3.4.0-RC1](https://github.com/spring-projects/spring-boot/releases/tag/v3.4.0-RC1)
* ✅ November 8: [Spring Cloud 2024.0.0-RC1](https://github.com/spring-cloud/spring-cloud-release/releases/tag/v2024.0.0-RC1)

Final releases
* ✅ November 14: [Spring Framework 6.2.0](https://github.com/spring-projects/spring-framework/releases/tag/v6.2.0)
* ✅ November 21: [Spring Boot 3.4.0](https://github.com/spring-projects/spring-boot/releases/tag/v3.4.0)
* ✅ December 3: [Spring Cloud 2024.0.0](https://github.com/spring-cloud/spring-cloud-release/milestone/147)

I expect that Spring Cloud GCP will require a new major version to support Spring Cloud 2024.0, Spring Boot 3.4 and Spring Framework 6.2. Do you agree?

To benefit for as long as possible from the support lifetime of Spring components, it would be great if a version of Spring Cloud GCP that supports these versions would be available not too long after these release dates. Do you already have any plans for releasing a version of Spring Cloud GCP that is compatible with these new Spring versions?

**Comment by burkedavison** [comment]
Yes, we will be supporting Spring Cloud 2024.0 shortly after its release.

> I expect that Spring Cloud GCP will require a new major version to support Spring Cloud 2024.0

Yes, we agree.

**Comment by CodingMaxima** [comment]
Thanks @breun for raising this. We're also waiting on this, thanks

**Comment by Fabio1988** [comment]
And I wondered why the GCP Secret Manager did not load the secret values anymore after updating Spring. Thanks, we are also waiting on this then :)

**Comment by maradanasai** [comment]
Thanks, we are also waiting on this :)

**Comment by breun** [comment]
Spring Framework 6.2.0, Spring Boot 3.4.0 and Spring Cloud 2024.0.0 are now all out.

**Comment by ccudennec-otto** [comment]
> And I wondered why the GCP Secret Manager did not load the secret values anymore after updating Spring. Thanks, we are also waiting on this then :)

@Fabio1988 we've also run into this and found @elg-entur's issue: #3391 . Just cross-linking if anyone's interested in the workaround.

**Comment by maradanasai** [comment]
waiting for spring cloud gcp's upgrade for Spring Cloud 2024.0.0

**Comment by ccudennec-otto** [comment]
Spring Cloud 2024.0.0 has been released: https://github.com/spring-cloud/spring-cloud-release/releases/tag/v2024.0.0

**Comment by csbiggar** [comment]
Also waiting eagerly for this compatibility :) do you have a proposed release date?

**Comment by syedtalha** [comment]
+1 Much needed.

**Comment by diegomarquezp** [comment]
We are actively working on providing support for Cloud 2024.0 in Spring Cloud GCP and will soon share the release timeline for this.

**Comment by csbiggar** [comment]
Are you able to share a release date for this yet? 🙏

**Comment by diegomarquezp** [comment]
We anticipate being able to release support shortly after the release of spring-boot 3.4.2 is out, [which is due by January 23rd](https://spring.io/projects#release-calendar).

**Comment by maradanasai** [comment]
@diegomarquezp SB 3.4.2 is out now :)

**Comment by csbiggar** [comment]
Are you able to share a release date now? 🙏

**Comment by nhmarujo** [comment]
Hi @diegomarquezp . What is your release cycle? Is there any calendar that we can consult for planned releases?

**Comment by maradanasai** [comment]
Hi @diegomarquezp Can you share the date for planned release for this ?

**Comment by ccudennec-otto** [comment]
[v6.0.0](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/releases/tag/v6.0.0) was published 🥳

**Comment by diegomarquezp** [comment]
Indeed, [6.0.0](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/releases/tag/v6.0.0) is out! I'm closing this for now. If any issues are found with this new release, please file them in our [issues section](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/issues).
